File: C++/Medium/WrongAnswer/decode-the-slanted-ciphertext_8.cpp
class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        
        int n = encodedText.size();
        int cols = n / rows;
        int total = rows * cols;
        int rem = n - total;
        int c = 0;
        string res;
        for (int i = 0; i < cols; i++) {
            for (int j = 0; j < rows; j++) {
                if (i % 2 == 0) {
                    res += encodedText[c++];
                } else {
                    if (j == rows - 1) {
                        res += encodedText[c++];
                    } else {
                        res += ' ';
                    }
                }
            }
        }
        int i = 0;
        while (rem > 0) {
            res += encodedText[total + i];
            i += rows - 1;
            rem--;
        }
        return res;
    }
};

File: Java/Medium/RuntimeError/decode-the-slanted-ciphertext_8.java
class Solution {
    public String decodeCiphertext(String encodedText, int rows) {
        
        if (rows == 1) return encodedText;
        int cols = encodedText.length() / rows;
        int[][] matrix = new int[rows][cols];
        int i = 0, j = 0;
        int dir = 1;
        for (int k = 0; k < encodedText.length(); k++) {
            matrix[i][j] = encodedText.charAt(k);
            i += dir;
            if (i == 0 || i == rows - 1) dir *= -1;
            j++;
        }
        StringBuilder sb = new StringBuilder();
        for (int k = 0; k < cols; k++) {
            for (i = 0; i < rows; i++) {
                if (matrix[i][k] != 0) sb.append((char) matrix[i][k]);
            }
        }
        return sb.toString();
    }
}